### DISCUSSION TITLE
using types from my project as output type

### DISCUSSION THREAD

**Discussion by MancTwonk** [RobinTail/express-zod-api#972] [Q&A] [answered]
Sorry if this is in the documentation but i could find it.

Is it possible to set the type of output using Types from within my application

i.e. if i have a Type
```ts
export type Foo = {
  bar: string
}
```

How do I set that as the output type

```ts
  method: 'post',
  input: z.object({
    foo: z.string().optional(),
  }),
  output: z.object({}), // <----- HERE i.e. output = { data: { THIS OBJECT IS OF TYPE FOO }}
```

Thanks in advance

**Comment by RobinTail** [comment, marked as answer]
Hello @MancTwonk ,

the concept works exactly the opposite way.
You declare schemas using `zod` schemas and then you can convert those schemas to Typescript types for your application (see Client generation in the docs).

In particular case:

```ts
const fooSchema = z.object({
  bar: z.string(),
});
```

then
```ts
// ...
output: fooSchema,
```

when using the `defaultEndpointsFactory` the final positive response will be exactly
```ts
{ 
  status: "success",
  data: { 
    foo: { 
      bar: string 
    } 
  } 
}
```

This way types can be declared more precise and contain additional features, like refinements and transformations.

However, if you still need to convert Typescript types to `zod` schemas, check out [`zod`'s ecosystem](https://github.com/colinhacks/zod#ecosystem) for the corresponding solution.
